ocean_lib/ocean/ocean_assets.py

```diff
@@ -141,24 +141,77 @@ def build_data_tokens_list(
         self, services: list, deployed_erc20_tokens: list
     ) -> list:
         data_tokens = []
+        # (1-n) service per datatoken, 1 datatoken per service
         for erc20_token in deployed_erc20_tokens:
             for service in services:
```

Comment:
let's make a list comprehension, I think the current structure warrants it. E.g. datatokens = datatokens + [{... that dict} for service in services if s.address == dt.address]